### ISSUE TITLE
Modal not showing in createOptionForm

### ISSUE THREAD

**Issue by aliteleb** [laravel-filament/filament#11274]
### Package

filament/forms

### Package Version

v3.2

### Laravel Version

v10.10

### Livewire Version

v3.4.4

### PHP Version

PHP 8.1

### Problem description

Modal not showing in createOptionForm in select form.

### Expected behavior

Category modal should be showing up

### Steps to reproduce

```php
Forms\Components\Select::make('title')
    ->relationship('category', 'name')
    ->searchable()
    ->required()
    ->native(false)
    ->createOptionForm([
        Forms\Components\TextInput::make('name')
            ->required()
            ->maxLength(255),
    ]),
```

### Reproduction repository

https://github.com/aliteleb/filament-select-issue.git

### Relevant log output

```shell
no errors no response
```

https://github.com/filamentphp/filament/assets/36816109/903d8294-4a8c-4917-9dc2-57173e2cd013

**Comment by Zal-com** [comment]
Has this been fixed or is there a solution found ? I can't get over this issue :/

**Comment by aliteleb** [comment]
> Has this been fixed or is there a solution found ? I can't get over this issue :/

Yeah. It was my fault.
I found the issue. The problem was due to an incorrect name attribute in the form component.

Problem:
I mistakenly used title instead of category_id in the Select component.
Changing title to category_id resolved the issue.

Problem: `Forms\Components\Select::make('title')`
Solution: `Forms\Components\Select::make('category_id')`

I hope this helps you.
